src/include/wiredtiger.in

```diff
@@ -3682,6 +3704,502 @@ struct __wt_extractor {
+	 * handle operations.
+	 *
+	 * @errors
+	 *
+	 * @param file_handle the WT_FILE_HANDLE
+	 * @param session the current WiredTiger session
+	 * @param map a location in the mapped region unlikely to be used in the
+	 *    near future
+	 * @param length the length of the mapped region to discard
+	 */
+	int (*map_discard)(WT_FILE_HANDLE *file_handle,
+	    WT_SESSION *session, void *map, size_t length);
+
+	/*!
+	 * Preload part of a memory mapped file, based on the POSIX 1003.1
+	 * standard madvise.
+	 *
+	 * This method is not required, and should be set to NULL when not
+	 * supported by the file.
+	 *
+	 * This method may be called concurrently with other file system or
+	 * handle operations.
+	 *
+	 * @errors
+	 *
+	 * @param file_handle the WT_FILE_HANDLE
+	 * @param session the current WiredTiger session
+	 * @param map a location in the mapped region likely to be used in the
+	 *    near future
+	 * @param length the size of the mapped region to preload
+	 */
+	int (*map_preload)(WT_FILE_HANDLE *file_handle,
+	    WT_SESSION *session, const void *map, size_t length);
+
+	/*!
+	 * Unmap a memory mapped file, based on the POSIX 1003.1 standard
+	 * munmap.
+	 *
+	 * This method is not required, and should be set to NULL when not
+	 * supported by the file.
+	 *
+	 * This method may be called concurrently with other file system or
+	 * handle operations.
+	 *
+	 * @errors
+	 *
+	 * @param file_handle the WT_FILE_HANDLE
+	 * @param session the current WiredTiger session
+	 * @param mapped_region a pointer to the start of the mapped region
+	 * @param length the length of the mapped region
+	 * @param mapped_cookie any cookie set by the WT_FILE_HANDLE::map method
+	 */
+	int (*unmap)(WT_FILE_HANDLE *file_handle, WT_SESSION *session,
+	    void *mapped_region, size_t length, void *mapped_cookie);
+
+	/*!
+	 * Read from a file, based on the POSIX 1003.1 standard pread.
+	 *
+	 * This method may be called concurrently with other file system or
+	 * handle operations.
+	 *
+	 * @errors
+	 *
+	 * @param file_handle the WT_FILE_HANDLE
+	 * @param session the current WiredTiger session
+	 * @param offset the offset in the file to start reading from
+	 * @param len the amount to read
+	 * @param[out] buf buffer to hold the content read from file
+	 */
+	int (*read)(WT_FILE_HANDLE *file_handle,
+	    WT_SESSION *session, wt_off_t offset, size_t len, void *buf);
+
+	/*!
+	 * Return the size of a file.
+	 *
+	 * This method may be called concurrently with other file system or
+	 * handle operations.
+	 *
+	 * @errors
+	 *
+	 * @param file_handle the WT_FILE_HANDLE
+	 * @param session the current WiredTiger session
+	 * @param sizep the size of the file
+	 */
+	int (*size)(
+	    WT_FILE_HANDLE *file_handle, WT_SESSION *session, wt_off_t *sizep);
+
+	/*!
+	 * Make outstanding file writes durable and do not return until writes
+	 * are complete.
+	 *
+	 * This method is not required, and should be set to NULL when not
+	 * required by the file.
```

Comment:
Are there implications for durability in a filesystem that supports updates but doesn't support `sync`?  I would be inclined to say this is required unless the filesystem is read-only.